edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs

```diff
@@ -301,14 +301,27 @@ static string ToProxyLogLevel(LogEventLevel level)
             return proxyLevel;
         }
 
-        V1Volume GetVolume(Mount mount)
+        V1Volume GetVolume(KubernetesModule module, Mount mount)
         {
-            string name = KubeUtils.SanitizeDNSValue(mount.Source);
-            return this.ShouldUsePvc()
-                ? new V1Volume(name, persistentVolumeClaim: new V1PersistentVolumeClaimVolumeSource(name, mount.ReadOnly))
-                : new V1Volume(name, emptyDir: new V1EmptyDirVolumeSource());
-        }
+            string name = KubeUtils.SanitizeK8sValue(mount.Source);
+            string pvcName = KubeUtils.SanitizeK8sValue(module.Name) + name;
```

Comment:
Like the idea. Thankd Denis. 

My understanding volume share occurs only when the volume (PV) exists, i.e. customers have the volume created manually. For that, this PR didn't change the behavior. 

In case of storageclass, first we crate PVC and then PV, where PVC and PV will have the same name. If customer doesn't specify k8s attributes, the emptydir will be created. So there is no volume to share.